crypto/store/store_result.c

```diff
@@ -324,9 +300,16 @@ static EVP_PKEY *try_key_value_legacy(struct extracted_param_data_st *data,
         X509_SIG *p8 = NULL;
         PKCS8_PRIV_KEY_INFO *p8info = NULL;
 
-        /* See if it's an encrypted PKCS#8 and decrypt it */
+        /*
+         * See if it's an encrypted PKCS#8 and decrypt it.  Discard DER
+         * decoding errors, a failed decode only means we return "empty handed"
+         */
+        ERR_set_mark();
```

Comment:
Same as above - if expected_type is set we should not blindly clear all the errors, otherwise there will be no way how to get any reasonable error information for the user.